packages/ERTP/src/amountMath.js

```diff
@@ -186,7 +186,10 @@ function makeAmountMath(brand, mathHelpersName) {
         allegedBrand !== undefined,
         details`alleged brand is undefined. Did you pass an extent rather than an amount?`,
       );
-      mustBeSameStructure(brand, allegedBrand, 'Unrecognized brand');
+      assert(
+        brand === allegedBrand,
+        details`the brand in the allegedAmount in 'coerce' didn't match the amountMath brand`,
```

Comment:
You have both brands. Can you print them? When I was debugging, this kind of thing has been very helpful.

Reply:
There's nothing to print. They are objects with a method (which we cannot access) and no properties.

Reply:
Don't they have `getAllegedName()`? That's enough to be useful.

Reply:
Yes, but it will likely be remote, so we can't call it

Reply:
That's too bad. 

Any chance we can cheat and include the objects so the caller could catch the error and display it themselves? For once, it might be useful to include the opaque object in an error.

Reply:
Please don't pass access by throw. When you throw something, you know very little about who might catch it.

In any case, if you include the brands as substitution holes, e.g.

```js
details`the brand in the allegedAmount in 'coerce' didn't match the amountMath brand: ${brand} vs ${allegedBrand}`,
```
this will throw an error with the message
```js
`the brand in the allegedAmount in 'coerce' didn't match the amountMath brand: (an object) vs (an object)
See console for error data.`
```
This avoids even sending the stringification of these substitution values to the catcher by default, under the default assumption that we're willing to reveal the literal part of the `details` template literal, but only the `typeof` of their substitution value.

OTOH, to the console, we send the actual substitution values themselves, not a mere stringification of them. If you are in the privileged position of receiving what is sent to the console, for example because you provided the console, then you can receive these values directly. For these purposes, we consider this to be loosely equivalent to the privilege level of viewing the computation through a debugger.

Attn @michaelfig and I are working towards a reform in the distributed object system so that presences can carry description strings which will show up in the default builtin console behavior. So to see that, you merely need access to the textual output of the normal console, which is still a debug level of privilege.

Reply:
@erights, I read what you said above as indicating that stringifying the brands would be useless, and that including the objects would make them accessible to print on the console. OTOH, you also said they shouldn't be conveyed by a `throw`. 
That sounds like including the objects in the `details` object would satisfy both objectives, since the console gets access to  `details`, but obfuscates the values in the thrown message.   
Am I misunderstanding?

Reply:
Putting brand objects in details will not show anything useful currently. It will show something like `{ getAllegedName }` (non-presence) or `{ }` (presence)